analysis_options.yaml

```diff
@@ -157,7 +164,7 @@ linter:
     - prefer_is_not_empty
     - prefer_is_not_operator
     - prefer_iterable_whereType
-    # - prefer_mixin # Has false positives, see https://github.com/dart-lang/linter/issues/3018
+    # - prefer_mixin # DIFFERENT FROM FLUTTER/FLUTTER: enable when v2.1.7 of plugin_platform_interface is the oldest supported version (which makes MockPlatformInterfaceMixin a mixin class)
```

Comment:
What does this comment mean exactly? The only version of any package we support is the latest version.

Reply:
The `analyze_downgraded` check was pulling in a version of the package where `MockPlatformInterfaceMixin` wasn't a mixin class yet and hence analysis would fail on packages that were using it as a mixin. As soon as  the `analyze_downgraded` check pulls in version 2.1.7 or newer of this package, the lint can be enabled. I guess, we could force this by updating the pubspec.yamls of all packages that depend on plugin_platform_interface to:

```yaml
plugin_platform_interface: ^2.1.7
```

If it's okay to do that, I can attempt this in a follow-up PR.

Reply:
Yes, upping the dependencies to 2.1.7 would be fine.

(That's the only way `analyze_downgraded` would pull 2.1.7. It doesn't have its own logic, it just runs `pub downgrade` to get the oldest allowed package set.)

Reply:
Awesome, thanks for confirming. I will prepare a follow-up PR to do just that after this one lands.